src/server/saved_objects/service/lib/repository.js

```diff
@@ -289,13 +295,15 @@ export class SavedObjectsRepository {
    * @property {string} [options.sortOrder]
    * @property {Array<string>} [options.fields]
    * @property {string} [options.namespace]
+   * @property {object} [options.referencedBy] - { type, id }
```

Comment:
I think a better name for this would be `hasReference`. `referencedBy` implies to me that we're only only going to return objects that are referenced by this reference, when instead we're going to return all objects that have a matching reference.